### PR TITLE
feat(#3204): Autoconfigure H2 Test Resource

### LINKED ISSUE #3204: Auto-configure H2 test resource
**Description**
Configuration of H2 could be automatically provided in the H2DatabaseTestResource as we know exactly what are the parameters we are configuring in the H2 instance.

Properties set using these method has a high preference than the ones provided when using %test profile.

Discussion here: https://groups.google.com/forum/#!topic/quarkus-dev/XL-sET6HFfs

### PR DESCRIPTION
**Pull request by lordofthejars**
Fixes #3204

### CODE REVIEW

**Comment by lordofthejars** [comment]
Ok, there is a problem when running the pattern of extending a test and annotate it with `@SubtrateVM`. The problem is that since `H2DatabaseTestResource` returned properties have more priority than the ones set in `application.properties` it makes the test fail because the native tests require a PostgreSQL, the configuration is not overridden to the PostgeSQL instance so the test fails.

So maybe we should close this PR as it is too restrictive or we stop using the `extend` pattern in IT tests.

File: extensions/agroal/deployment/src/test/resources/application-default-datasource.properties [hunk 1]
```diff
@@ -1,5 +1,5 @@
 #tag::basic[]
-quarkus.datasource.url=jdbc:h2:tcp://localhost/mem:default
+quarkus.datasource.url=jdbc:h2:mem:default
```
Comment (gsmet) [inline extensions/agroal/deployment/src/test/resources/application-default-datasource.properties]: What's the rationale of this change? I don't see how it is an improvement?

**Review by gsmet — CHANGES_REQUESTED** [review]
Blocking the PR for now, we need some clarification about the changes applied.

**Comment by lordofthejars** [comment]
@gsmet https://groups.google.com/forum/#!topic/quarkus-dev/XL-sET6HFfs

But as I said now after running the IT tests, I don't know if it is not too restrictive.

File: extensions/agroal/deployment/src/test/resources/application-default-datasource.properties (hunk 1 above)
Comment (gsmet) [reply extensions/agroal/deployment/src/test/resources/application-default-datasource.properties]: I have read the thread.

Still don't see why we change all these URLs everywhere?

Comment (lordofthejars) [reply extensions/agroal/deployment/src/test/resources/application-default-datasource.properties]: Ah ok sorry, you are changing because in these tests you need more than one different DataSource so since now `H2DatabaseTestResource` can only provide one H2 (because the config is hardcoded) I cannot use it in these tests and I need to create an embedded memory database. This comes because of this limitation of hardcoding properties, then you cannot use it in these scenarions

Comment (gsmet) [reply extensions/agroal/deployment/src/test/resources/application-default-datasource.properties]: Yeah, frankly, considering this limitation, I don't see this as an improvement.

Comment (lordofthejars) [reply extensions/agroal/deployment/src/test/resources/application-default-datasource.properties]: Exactly this was what I have noticed after seeing the IT failing tests, but since I have already sent the PR I wanted to just discuss. So I think that for now, we can close it.

**Comment by aloubyansky** [comment]
Ok, closing this PR. Thanks for researching it @lordofthejars
